exchanges/bitfinex/bitfinex.go

```diff
@@ -110,16 +118,265 @@ func (b *Bitfinex) GetPlatformStatus() (int, error) {
 	return -1, fmt.Errorf("unexpected platform status value %d", response[0])
 }
 
+// GetV2MarginFunding gets borrowing rates for margin trading
+func (b *Bitfinex) GetV2MarginFunding(symbol, amount string, period int32) (MarginV2FundingData, error) {
+	var resp []interface{}
+	var response MarginV2FundingData
+	params := make(map[string]interface{})
+	params["symbol"] = symbol
+	params["period"] = period
+	params["amount"] = amount
+	err := b.SendAuthenticatedHTTPRequestV2(exchange.RestSpot, http.MethodPost,
+		bitfinexV2MarginFunding,
+		params,
+		&resp,
+		getAccountFees)
+	if err != nil {
+		return response, err
+	}
+	if len(resp) != 2 {
+		return response, errors.New("invalid data received")
+	}
+	avgRate, ok := resp[0].(float64)
+	if !ok {
+		return response, errors.New("failed type assertion for rate")
+	}
+	avgAmount, ok := resp[1].(float64)
+	if !ok {
+		return response, errors.New("failed type assertion for amount")
+	}
+	response.Symbol = symbol
+	response.RateAverage = avgRate
+	response.AmountAverage = avgAmount
+	return response, nil
+}
+
+// GetV2FundingInfo gets funding info for margin pairs
+func (b *Bitfinex) GetV2FundingInfo(key string) (MarginFundingDataV2, error) {
+	var resp []interface{}
+	var response MarginFundingDataV2
+	err := b.SendAuthenticatedHTTPRequestV2(exchange.RestSpot, http.MethodPost,
+		fmt.Sprintf(bitfinexV2FundingInfo, key),
+		nil,
+		&resp,
+		getAccountFees)
+	if err != nil {
+		return response, err
+	}
+	if len(resp) != 3 {
+		return response, errors.New("invalid data received")
+	}
+	sym, ok := resp[0].(string)
+	if !ok {
+		return response, errors.New("failed type assertion for sym")
+	}
+	symbol, ok := resp[1].(string)
+	if !ok {
+		return response, errors.New("failed type assertion for symbol")
+	}
+	fundingData, ok := resp[2].([]interface{})
+	if !ok {
+		return response, errors.New("failed type assertion for fundingData")
+	}
+	response.Sym = sym
+	response.Symbol = symbol
+	for x := 0; x < 3; x++ {
+		_, ok := fundingData[x].(float64)
+		if !ok {
+			return response, fmt.Errorf("type conversion failed for x = %d", x)
+		}
+	}
+	response.Data.YieldLoan = fundingData[0].(float64)
+	response.Data.YieldLend = fundingData[1].(float64)
+	response.Data.DurationLoan = fundingData[2].(float64)
+	response.Data.DurationLend = fundingData[3].(float64)
+	return response, nil
+}
+
+// GetAccountInfoV2 gets V2 account data
+func (b *Bitfinex) GetAccountInfoV2() (AccountV2Data, error) {
+	var resp AccountV2Data
+	var data []interface{}
+	err := b.SendAuthenticatedHTTPRequestV2(exchange.RestSpot, http.MethodPost,
+		bitfinexV2AccountInfo,
+		nil,
+		&data,
+		getAccountFees)
+	if err != nil {
+		return resp, err
+	}
+	var ok bool
+	var tempString string
+	var tempFloat float64
+	if tempFloat, ok = data[0].(float64); !ok {
```

Comment:
Please gate this section with a length check to avoid panics
```
func TestThing(t *testing.T) {
	data := []interface{}{}
	if tempFloat, ok := data[0].(float64); !ok {
		t.Log(tempFloat)
	}
}
```